src/fiatconnect/saga.ts

```diff
@@ -69,17 +80,116 @@ export function* handleFetchFiatConnectQuotes({
       flow,
       fiatConnectCashInEnabled,
       fiatConnectCashOutEnabled,
-      fiatConnectProviders,
+      fiatConnectProviders: providerIds
+        ? fiatConnectProviders.filter(({ id }) => providerIds.includes(id))
+        : fiatConnectProviders,
     })
     yield put(fetchFiatConnectQuotesCompleted({ quotes }))
   } catch (error) {
     Logger.error(TAG, 'Could not fetch fiatconnect quotes', error)
     yield put(fetchFiatConnectQuotesFailed({ error: 'Could not fetch fiatconnect quotes' }))
   }
 }
+/**
+ * This saga attempts to fetch a quote and matching fiatAccount for a provider that the user
+ * has previously used in the CICO flow.
+ * If successful: Navigates the user to the FiatConnectReview screen
+ * If fail: Navigates the user to the SelectProvider screen
+ */
+export function* handleAttemptReturnUserFlow({
+  payload: params,
+}: ReturnType<typeof attemptReturnUserFlow>) {
+  const { amount, flow, selectedCrypto, providerId, fiatAccountId, fiatAccountType } = params
+  const digitalAsset = {
+    [Currency.Celo]: CiCoCurrency.CELO,
+    [Currency.Dollar]: CiCoCurrency.CUSD,
+    [Currency.Euro]: CiCoCurrency.CEUR,
+  }[selectedCrypto]
+  const fiatConnectProviders: FiatConnectProviderInfo[] | null = yield select(
+    fiatConnectProvidersSelector
+  )
+  try {
+    // Fetch Quote associated with the cached providerId
+    yield put(
+      fetchFiatConnectQuotes({
+        digitalAsset,
+        cryptoAmount: amount.crypto,
+        flow,
+        providerIds: [providerId],
+      })
+    )
+    const quotes: (FiatConnectQuoteSuccess | FiatConnectQuoteError)[] = yield select(
+      fiatConnectQuotesSelector
+    )
+    const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
+    const normalizedQuote = normalizedQuotes.find(
+      (q) => q.getFiatAccountType() === fiatAccountType && q.getProviderId() === providerId
+    )
+    if (!normalizedQuote) {
+      throw new Error('Could not find quote')
+    }
+    // Get the provider info
+    const fiatConnectProvider = fiatConnectProviders?.find((provider) => provider.id === providerId)
+    if (!fiatConnectProvider) {
+      throw new Error('Could not find provider')
+    }
+    // Fetch Fiat Account associated with the cached providerId / fiatAccountId
+    const fiatAccounts: FiatAccount[] = yield call(
+      fetchFiatAccountsSaga,
+      providerId,
+      fiatConnectProvider.baseUrl
+    )
+    const fiatAccount = fiatAccounts.find((account) => account.fiatAccountId === fiatAccountId)
+    if (!fiatAccount) {
+      throw new Error('Could not find fiat account')
+    }
 
-export function* watchFetchFiatConnectQuotes() {
-  yield takeLeading(fetchFiatConnectQuotes.type, handleFetchFiatConnectQuotes)
+    // Successfully found quote and fiatAccount
+    yield put(attemptReturnUserFlowCompleted())
+    navigate(Screens.FiatConnectReview, {
+      flow,
+      normalizedQuote,
+      fiatAccount,
+    })
+  } catch (error) {
```

Comment:
Should we handle transient errors like n/w failure and provider / fiat account mismatch errors differently? Seems like at a minimum we'd want to log warn / errors for n/w errors.

Reply:
What would be the goal of making them errors? How would you suggest we categorize the different failure types?

Reply:
hmm maybe we can track them? in sentry for example? If we want to track provider request success rate, etc.
We can use a custom error for errors thrown in the try block and do different actions.

Reply:
I thought that one of the directions we were going with Sentry was to *not* use it for transient errors like a partner failure?

Reply:
chatted offline, current implementation is good.